read.go

```diff
@@ -169,6 +174,27 @@ func (r *reader) readTOC(toc *indexTOC) error {
 	return nil
 }
 
+func (r *reader) readHeader() (simpleSection, uint32, error) {
```

Comment:
I factored out the first part of `readTOC` (now `readTOCSections`). This wasn't critical for the change.